plugins/modules/proxmox_kvm.py

```diff
@@ -1323,18 +1323,18 @@ def main():
             module.fail_json(vmid=vmid, msg='Unable to migrate VM {0} from {1} to {2}: {3}'.format(vmid, vm_node, node, e))
 
     if state == 'present':
-        try:
-            if proxmox.get_vm(vmid, ignore_missing=True) and not (update or clone):
-                module.exit_json(changed=False, vmid=vmid, msg="VM with vmid <%s> already exists" % vmid)
-            elif proxmox.get_vmid(name, ignore_missing=True) and not (update or clone):
-                module.exit_json(changed=False, vmid=proxmox.get_vmid(name), msg="VM with name <%s> already exists" % name)
-            elif not node:
-                module.fail_json(msg='node is mandatory for creating/updating VM')
-            elif update and not any([vmid, name]):
-                module.fail_json(msg='vmid or name is mandatory for updating VM')
-            elif not proxmox.get_node(node):
-                module.fail_json(msg="node '%s' does not exist in cluster" % node)
+        if not (update or clone) and proxmox.get_vm(vmid, ignore_missing=True):
```

Comment:
I moved those checks outside of the try/catch block as they couldn't cause any exceptions.

Reply:
`proxmox.get_vm()` calls `self.proxmox_api.cluster.resources.get(type='vm')`, which I guess could result in exceptions (for example because of connection errors while talking with the API)?

Reply:
Oh.. right. I think those function body needs to be wrapped into try/catch.
Otherwise, I couldn't solve the issue with unit testing. When AnsibleExitJson raises it causes to be caught by catch block and throws another AnsibleFailJson, which then hides kwargs for exit_json call and I can't test it. :-(
Maybe there is another possible solution, I couldn't find it yet.

```
raise AnsibleFailJson(kwargs)
E       ansible_collections.community.general.tests.unit.plugins.modules.utils.AnsibleFailJson: {'vmid': 100, 'msg': "creation of qemu VM None with vmid 100 failed with exception={'changed': False, 'vmid': 100, 'msg': 'VM with vmid <100> already exists'}", 'failed': True}
```